### PR TITLE
Fix bug that shape mismatch error in predict when changing objective to softmax and update rust-xgboost commit

### PR DESCRIPTION
**Pull request by cyccbxhl**
Fix bugs about rust-xgboost:
        1) NaN recall;
        2) Shape mismatch error in predict when changing objective to softmax.

### CODE REVIEW

**Comment by montanalow** [comment]
It would be nice if XGBooster had a get_param, so we didn't need to pass hyperparams around like this, but it doesn't, so we do. Thanks for fixing this up.
